pkg/operator/ceohelpers/unsupported_override.go

```diff
@@ -0,0 +1,29 @@
+package ceohelpers
+
+import (
+	"bytes"
+	"encoding/json"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+)
+
+// IsUnsupportedUnsafeEtcd returns true if
+// useUnsupportedUnsafeNonHANonProductionUnstableEtcd key is set
+// to any parsable value
+func IsUnsupportedUnsafeEtcd(spec *operatorv1.StaticPodOperatorSpec) (bool, error) {
+	unsupportedConfig := map[string]interface{}{}
+	if spec.UnsupportedConfigOverrides.Raw == nil {
+		return false, nil
+	}
+	if err := json.NewDecoder(bytes.NewBuffer(spec.UnsupportedConfigOverrides.Raw)).Decode(&unsupportedConfig); err != nil {
+		klog.V(4).Infof("decode of unsupported config failed with error: %v", err)
+		return false, err
+	}
+
+	_, found, err := unstructured.NestedFieldNoCopy(unsupportedConfig, "useUnsupportedUnsafeNonHANonProductionUnstableEtcd")
```

Comment:
Add a comment explaining why this is unsafe.  I think the answer is

1. this violates operational best practices for etcd - unstable
2. this allows non-HA configurations which we cannot support in production - unsafe and non-HA
3. this allows a situation where we can get stuck unable to re-achieve quorum, resulting in cluster-death - unsafe, non-production, unstable
4. The combination of all these things makes the situation unsupportable.

I think clearly communicating these reasons for choosing such a long name makes the consequences of this choice clear to all consumers.  The name isn't offensive, it is descriptive for the state it  places etcd in.

Reply:
added, thanks for drafting the reasons.